### PR TITLE
Add all data set types to quality page

### LINKED ISSUE #4151: [Dataset quality] Opening page for other dataStream types
### Description

We released Dataset Quality page for dataStreams of type logs, specially the ones following DSNS.
Recently we started working in enabling the page for `traces`, `metrics` and `synthetics` dataStreams. (check out the [project](https://github.com/elastic/observability-dev/issues/3505) for more information)

### Resources

This feature is part of [[project] Dataset quality page for all data](https://github.com/elastic/observability-dev/issues/3505).
The changes are tracked in the issues of the project.

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

The feature is identical in both

### What release is this request related to?

8.16

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @yngrdyn 

**Stakeholders:** @gbamparop

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
This PR closes #4151.

The Data Set Quality page can now show data sets from metrics, traces, and synthetics in addition to logs. 
I've update the page to show this and updated the screenshot.

### CODE REVIEW

**Review by mdbirnstiehl — COMMENTED** [review]
@yngrdyn will the steps to investigate ignored fields detailed [here ](https://github.com/elastic/observability-docs/pull/4232/files#diff-5835373ce7a405eb814e05dbff2e2e5151aa3ce4859ba1a4471a3e492b6821fcR33-R70) work for the other data set types at this point?

**Comment by yngrdyn** [comment]
> @yngrdyn will the steps to investigate ignored fields detailed [here ](https://github.com/elastic/observability-docs/pull/4232/files#diff-5835373ce7a405eb814e05dbff2e2e5151aa3ce4859ba1a4471a3e492b6821fcR33-R70)work for the other data set types at this point?

yes, It works mostly the same. The only thing is you won't be able to spot dataset quality issues from logs explorer for these other types. User will find the hints in discover.
And from the dataset quality page they will need to enable the new types using the `type` filter as per default it will only bring logs.

**Comment by mdbirnstiehl** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-kj4sevcqd-elastic-dev.vercel.app

* Built with commit 7cf5a779cb10d716905f4c1ab72818864bb7f154

> Issues? Visit #next-docs in Slack

**Comment by mdbirnstiehl** [comment]
@yngrdyn could you take a look at this one?

**Review by yngrdyn — DISMISSED** [review]
LGTM

File: docs/en/observability/logs-monitor-datasets.asciidoc [hunk 1]
```diff
@@ -51,19 +53,19 @@ The **Quality issues** section shows fields that were ignored during ingest, the
 
 [discrete]
 [[find-ignored-fields-in-individual-logs]]
-=== Find ignored fields in individual logs
+=== Find ignored fields in individual documents
 
-To use Logs Explorer to find ignored fields in individual logs:
+To use Logs Explorer or discover to find ignored fields in individual documents:
```
Comment (colleenmcginnis) [inline docs/en/observability/logs-monitor-datasets.asciidoc:58]: Should this be capitalized?

```suggestion
To use Logs Explorer or Discover to find ignored fields in individual documents:
```

**Review by colleenmcginnis — COMMENTED** [review]
Did you intend for this to be a new top-level item in the Observability guide? (I'm not saying it shouldn't be because I don't know where else it would go, but I'm just checking!)

Screenshot 2024-09-25 at 9 47 51 AM

**Comment by mdbirnstiehl** [comment]
> Did you intend for this to be a new top-level item in the Observability guide? (I'm not saying it shouldn't be because I don't know where else it would go, but I'm just checking!)
> 
> Screenshot 2024-09-25 at 9 47 51 AM

I did, but I also meant to ask @dedemorton for any suggestions she might have on this since it is the beautiful new IA, I don't want to muddy it up already.

File: docs/en/observability/logs-monitor-datasets.asciidoc [hunk 2]
```diff
@@ -1,14 +1,16 @@
 [[monitor-datasets]]
-= Monitor log data set quality
+= Monitor data set quality
```
Comment (colleenmcginnis) [inline docs/en/observability/logs-monitor-datasets.asciidoc:2]: Maybe remove `Monitor` to align with other top-level items in the table of contents? 

```suggestion
= Data set quality
```

**Review by colleenmcginnis — DISMISSED** [review]
✨ 🐥

**Comment by bmorelli25** [comment]
Should this go into 8.x?

File: docs/en/serverless/monitor-datasets.mdx [hunk 3]
```diff
@@ -1,18 +1,18 @@
 ---
 id: serverlessObservabilityMonitorDatasets
 slug: /serverless/observability/monitor-datasets
-title: Monitor log data set quality
-description: Monitor log data sets to find degraded documents.
+title: Monitor data set quality
```
Comment (mdbirnstiehl) [inline docs/en/serverless/monitor-datasets.mdx:4]:
```suggestion
title: Data set quality monitoring
```

**Review by colleenmcginnis — APPROVED** [review]
🦭 👍

**Pull request merged**